{"id": "1005889", "question": "pandas的groupby报错：ValueError: Length of values does not match length of index", "description": "<div class=\"col-md-11 col-xs-10\"><p>感觉是发现了pandas的bug，同样的dataframe，对一列groupby的什么没问题，同时对两列groupby的时候就报错了</p><pre><code class=\"python\">#不会报错\ndata.groupby('key1', as_index=False).agg({'y': 'mean'})\n\n#不会报错\ndata.groupby('key2', as_index=False).agg({'y': 'mean'})\n\n#报错 ValueError: Length of values does not match length of index\ndata.groupby(['key1', 'key2'], as_index=False).agg({'y': 'mean'})</code></pre><p>摸不着头脑，不知道是什么问题</p></div>", "viewer": 8831, "tags": ["统计/机器学习", "Python"], "answers": [{"lvl1_answer": "<div class=\"col-md-11 col-xs-10 p-r\"><p>试试</p><pre><code class=\"python\">data.groupby(['key1', 'key2']).agg({'y': 'mean'}).reset_index()</code></pre><p>我以前也遇到过这个问题，后来就是这么解决的，不知道对你有没有帮助</p></div>", "lvl2_answer": ["神了，这样子果然可以了！NB！"]}]}